### ISSUE TITLE
Round corners

### ISSUE THREAD

**Issue by stsdc** [lcallarec/live-chart#27]
I don't know if this is a limitation of Gtk3 or Cairo, but I'm unable to create proper round corners.

![image](https://user-images.githubusercontent.com/6031763/102620801-e2931580-413e-11eb-8eb9-2d0b1624a20b.png)

As You can see on screenshot it is not properly cropped. :confused: 

[Alain managed to create round corners](https://github.com/stsdc/monitor/pull/210), but it was only possible when paddings were added.

**Comment by lcallarec** [comment]
Yes, AFAIK it's a common issue with most gtk widgets.

You could try this :

* Remove the chart's background : https://github.com/stsdc/monitor/blob/master/src/Widgets/Chart/Chart.vala#L30
* Hide the whole background with `live_chart.background.visible = false`
* Set the white background color with CSS in your `.graph` class
* Remove the paddings

But this solution has a drawback : chart's lines will still overflow the corners if they cross... But maybe it's gona be good enough for your case ?

If it's not, I'll fix that in LiveChart itself...

**Comment by stsdc** [comment]
Yeah, it works. But line overlaps a border line:( I presume grid will too.
![image](https://user-images.githubusercontent.com/6031763/102699313-f83e3300-4243-11eb-8bf8-5bf9a862f34c.png)

**Comment by stsdc** [comment]
Here it is almost perfect:
![image](https://user-images.githubusercontent.com/6031763/102699888-b4016180-4248-11eb-8cfb-962f85833757.png)

**Comment by lcallarec** [comment]
Do you consider that's good enough and nothing should be done on livechart side ?

**Comment by stsdc** [comment]
I mean, this is more like a wish 😅

it would be nice have the line cropped. If the line will thicker it will be more noticeable. 

If you will consider this out of scope because it should be supported by Gtk or Cairo, I'll understand.

**Comment by lcallarec** [comment]
I'll take a look about adding rounded corners, but not sure it's gonna work...

**Comment by stsdc** [comment]
[This may be related](https://www.reddit.com/r/GTK/comments/kgsofe/graphs_in_gtk/)

**Comment by lcallarec** [comment]
Hi! @stsdc ! Just for information, I'll have some free time to work on this issue in the next weeks. Stay tuned :)

**Comment by stsdc** [comment]
Yay! ☺️

Just FYI, There are rumors that overflow issues might be fixed in GTK4. I don't know if that's the case for a cairo based widget, tho.

**Comment by stsdc** [comment]
So, GTK4 didn't fix the issue.
Maybe this solution will be handy: https://github.com/ensemblesaw/ens-shell-gtk/blob/layout/src/Classes/Widgets/Display/EqualizerBar.vala#L58
